src/main/scala/mesosphere/marathon/core/storage/StorageConfig.scala

```diff
@@ -29,10 +29,11 @@ import scala.collection.immutable.Seq
 import scala.concurrent.ExecutionContext
 import scala.concurrent.duration.{ Duration, _ }
 import scala.reflect.ClassTag
+// scalastyle:on
 
 sealed trait StorageConfig
```

Comment:
1)
In other modules, the config is a trait, which defines the configuration properties of this module.
The `StorageConfig` totally makes sense, but is something different. 
The `StorageModule` takes the complete `MarathonConf` - so it is hard to see which configuration parameter will be used (and it is hard to use those classes in another context e.g. metronome).
Perhaps it makes sense to move this class into `impl`?
Can you create a trait that defines the config options of this module?

2) 
I would like to see the repository as a separate top level module.
This component could be used from other applications (like metronome) and should not contain specific classes (like GroupRepositoryImpl)